src/graphdb.js

```diff
@@ -52,74 +52,157 @@ function toPascalCase (str) {
+
+                let fromCase = replaceCleanseLabel(direction.from);
+                let invalidDirFrom = direction.from !== fromCase ? true : false;
+                fromCase = toPascalCase(fromCase);
+
+                let toCase = replaceCleanseLabel(direction.to);
+                let invalidDirTo = direction.to !== toCase ? true : false;
+                toCase = toPascalCase(toCase);
+
+                let edgeCase = replaceCleanseLabel(edge.label);
+                let invalidEdge = edge.label !== edgeCase ? true : false;
+                edgeCase = toPascalCase(edgeCase);
+
                 if (direction.from == node.label && direction.to == node.label){
                     if (direction.relationship == 'MANY-MANY') {
-                        r += `\t${nodeCase.toLocaleLowerCase() + edgeCase}sOut(filter: ${nodeCase}Input, options: Options): [${nodeCase}] @relationship(edgeType:"${edge.label}", direction:OUT)\n`;
-                        r += `\t${nodeCase.toLocaleLowerCase() + edgeCase}sIn(filter: ${nodeCase}Input, options: Options): [${nodeCase}] @relationship(edgeType:"${edge.label}", direction:IN)\n`;                    
+                        if (invalidNode || invalidEdge) {
+                            r += `\t${nodeCase.toLocaleLowerCase() + edgeCase}sOut(filter: ${nodeCase}Input, options: Options): [${nodeCase}] @relationship(edgeType:"${edge.label}", direction:OUT) @alias(property: "${node.label.toLocaleLowerCase() + toPascalCase(edge.label)}sOut")\n`;
+                            r += `\t${nodeCase.toLocaleLowerCase() + edgeCase}sIn(filter: ${nodeCase}Input, options: Options): [${nodeCase}] @relationship(edgeType:"${edge.label}", direction:IN) @alias(property: "${node.label.toLocaleLowerCase() + toPascalCase(edge.label)}sIn")\n`;
+                        }
+                        else {
+                            r += `\t${nodeCase.toLocaleLowerCase() + edgeCase}sOut(filter: ${nodeCase}Input, options: Options): [${nodeCase}] @relationship(edgeType:"${edge.label}", direction:OUT)\n`;
+                            r += `\t${nodeCase.toLocaleLowerCase() + edgeCase}sIn(filter: ${nodeCase}Input, options: Options): [${nodeCase}] @relationship(edgeType:"${edge.label}", direction:IN)\n`;
+                        }
                     }
                     if (direction.relationship == 'ONE-ONE') {
-                        r += `\t${nodeCase.toLocaleLowerCase() + edgeCase}Out: ${nodeCase} @relationship(edgeType:"${edge.label}", direction:OUT)\n`;
-                        r += `\t${nodeCase.toLocaleLowerCase() + edgeCase}In: ${nodeCase} @relationship(edgeType:"${edge.label}", direction:IN)\n`;
+                        if (invalidNode || invalidEdge) {
+                            r += `\t${nodeCase.toLocaleLowerCase() + edgeCase}Out: ${nodeCase} @relationship(edgeType:"${edge.label}", direction:OUT) @alias(property: "${node.label.toLocaleLowerCase() + toPascalCase(edge.label)}Out")\n`;
```

Comment:
Are aliases for relationships used in the resolver templates? I don't see the alias value being used to generate the open cypher queries if there is a relationship directive?

Reply:
If I comment out the code that adds `@alias` to everything with a `@relationship` the queries still work for me
![Screenshot 2024-11-06 at 4 48 39 PM](https://github.com/user-attachments/assets/19cc9b99-d5c5-4f3f-9be9-1ceed1b5c029)

Reply:
if im understanding this right, i don't think the relationships part ex. `@relationship(edgeType:"${edge.label}"` uses aliases so i don't think it's being used? i think the aliases part are just used for the labels

Reply:
I don't see the alias added after the `@relationship` being used at all?

Reply:
yeah you're right aliases aren't being used in queries with `@relationship` directives. i finished removing `@alias` as you suggested 👍